sdk/mx.sdk/mx_sdk_vm_impl.py

```diff
@@ -1113,8 +1114,8 @@ def __init__(self, component, image_config, **kw_args):
         :type component: mx_sdk.GraalVmComponent | None
         :type image_config: mx_sdk.AbstractNativeImageConfig
         """
-        deps = []
         self.image_config = image_config
+        deps = list(image_config.jar_distributions) if _get_svm_support().is_supported() else []
```

Comment:
Why is this about SVM support? shouldn't it rather be about 8 vs 11+?

Reply:
I took inspiration from `mx_sdk_vm_impl.GraalVmNativeImage.__init__`
```python
svm_support = _get_svm_support()
if svm_support.is_supported():
    deps += self.native_image_jar_distributions
```
Previously I just had
```
deps = self.native_image_jar_distributions
```
here, but seeing the code from `mx_sdk_vm_impl.GraalVmNativeImage.__init__` made me suspect that `svm_support.is_supported()`should also be used here.

Reply:
The reason why I need to specify the dependencies is that I need to run the `NativePropertiesBuildTask` after the jars are made into proper modules so that using
```python
build_args += image_config.get_add_exports(set(export_deps_to_exclude))
```
in `mx_sdk_vm_impl.NativePropertiesBuildTask.contents` works reliably.

Reply:
I think it would be worth explaining why that dependency is added in a comment.

Regarding the check I think `GraalVmNativeImage` checks `_get_svm_support().is_supported()` because if that's true then the native image will be built which requires the input jars to exist. That's because those projects can either build native images or just shell launchers.
On the other end, `GraalVmNativeProperties` only exists if SVM is supported anyway so i don't think this check adds much. However a version check would remove the dependency when it's not necessary (on 8)

Reply:
> However a version check would remove the dependency when it's not necessary (on 8)

Makes sense. I will add that.

Reply:
Fixed in fdae21edcc566b12bd00e989302c56d86217c478